src/sqlBible/src/models/BibleCrossReference.entity.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    Entity,
+    Column,
+    PrimaryGeneratedColumn,
+    ManyToOne,
+    Index,
+    AfterLoad,
+    BeforeInsert
+} from 'typeorm';
+import { parsePhraseId, generateNormalizedRefId } from '../utils';
+import {
+    BiblePhrase,
+    BibleSection,
+    IBibleReferenceRangeVersion,
+    IBibleReferenceRangeNormalized
+} from '.';
+
+@Entity()
+export class BibleCrossReference
+    implements IBibleReferenceRangeVersion, IBibleReferenceRangeNormalized {
+    @PrimaryGeneratedColumn()
+    id?: number;
+
+    @Column()
+    normalizedRefId: number;
+
+    @Column({ nullable: true })
+    normalizedRefIdEnd?: number;
+
+    // the normalizedRefIds encode the following attributes:
+    bookOsisId: string;
```

Comment:
i usually also prefer to do that. however in some cases it can improve readability a lot if you group attributes differently. I feel this is the case here with the combination of TypeORM, a related sql schema and class attributes that depend on others via encoding / decoding.